python/revng/cli/_commands/fetch_debuginfo/elf.py

```diff
@@ -0,0 +1,195 @@
+
+            log(f"Could not find {debug_file_name}...", verbose)
+            # TODO: Should we `return None` here?
+        else:
+            log("error: Unable to parse of debug info file from .gnu_debuglink", verbose)
+            # TODO: Should we `return None` here?
+
+    build_id_dir = global_debug_directory + ".build-id/"
+    if not os.path.exists(build_id_dir):
+        log(f"Could not find {build_id_dir}", verbose)
+        return None
+
+    # 3) Build ID.
+    #  If build-id is `abcdef1234`, we look for:
+    #   - /usr/lib/debug/.build-id/ab/cdef1234.debug
+    log("Parsing build-id...", verbose)
+    biuld_id = parse_build_id(file)
+    if biuld_id is None:
+        # If we cannot parse the build id, we cannot fetch debug info.
+        return None
+
+    log("BUILD-ID: " + biuld_id, verbose)
+    # NOTE: The first two bytes form a directory inside `/usr/lib/debug/.build-id/` and
+    # rest of the hash is used for name of the debug info file.
+    first_two_chars = biuld_id[:2]
+    the_rest_of_build_hash = biuld_id[2:]
+    debug_file_name = the_rest_of_build_hash + ".debug"
+    absolute_path_debuginfo = os.path.join(build_id_dir, first_two_chars, debug_file_name)
+    if os.path.exists(absolute_path_debuginfo):
+        log("Debug File found: " + absolute_path_debuginfo, verbose)
+        return absolute_path_debuginfo
+    else:
+        log(f"Could not find {absolute_path_debuginfo}", verbose)
+
+    # 4) Find debug info on the web as `debuginfod` does.
+    if "XDG_CACHE_HOME" not in os.environ or os.environ["XDG_CACHE_HOME"] == "":
+        log("warning: XDG_CACHE_HOME was not set. Using ~/.cache/debuginfod_client/...", verbose)
+        path_to_download = str(Path.home()) + "/.cache/debuginfod_client/" + biuld_id + "/"
+    else:
+        path_to_download = os.environ["XDG_CACHE_HOME"] + "/debuginfod_client/" + biuld_id + "/"
+
+    if not os.path.exists(path_to_download):
+        os.makedirs(path_to_download)
+
+    debug_file_to_download = os.path.join(path_to_download, "debug")
+    if os.path.exists(debug_file_to_download):
+        log("Already downloaded debug file from web...", verbose)
+        return debug_file_to_download
+
+    log("Trying to find the debug info on the web...", verbose)
+    for url in urls:
+        debug_info_url = url + "/buildid/" + biuld_id + "/debuginfo"
+        log(f"Trying to download from {debug_info_url}...", verbose)
+        if download_file(debug_info_url, debug_file_to_download, verbose) is True:
+            # TODO: This can be compressed, if the file was compressed. Do we need to uncopress it?
+            return debug_file_to_download
+
+    # 5) Find debug symbols with Ubuntu Debug Symbol Packages.
+    log("Trying to find on the web with apt-file...", verbose)
+    if check_dbgsym_packages(verbose):
+        if shutil.which("dpkg") is not None:
+            proc = subprocess.run(["dpkg", "-S", file_path], capture_output=True)
```

Comment:
* Check the exit code of the program
* Fix the comment, we're not using `apt-file`.